sunpy/coordinates/frames.py

```diff
@@ -59,14 +64,17 @@ def __str__(self):
             return super().__str__()
 
 
+@add_common_docstring(**_variables_for_parse_time_docstring())
 class HeliographicStonyhurst(SunPyBaseCoordinateFrame):
     """
     A coordinate or frame in the Stonyhurst Heliographic system.
 
     In a cartesian representation this is also known as the Heliocentric
     Earth Equatorial (HEEQ) system. This frame has its origin at the solar
     centre and the north pole above the solar north pole, and the zero line on
-    longitude pointing towards the Earth.
+    longitude pointing towards the Earth. If the ``data`` parameter
+    is given, the positional parameters for the coordinate frame
+    (lon, lat, radius) do not need to be given.
 
     A new instance can be created using the following signatures
```

Comment:
Add a block like this to `Heliocentric` to show both the default (Cartesian) instantiation as well as the optional cylindrical instantiation.

Reply:
Upon some playing around, I also think the location parameters (ex. x, y, z or lon, lat, radius) are positional parameters and don't have to be keyword parameters. I edited the docstrings on HeliographicStonyhurst and Heliocentric accordingly.